src/audio/tdfb/tdfb.c

```diff
@@ -511,10 +511,10 @@ static int tdfb_setup(struct processing_module *mod, int source_nch, int sink_nc
 
 	if (delay_size > cd->fir_delay_size) {
 		/* Free existing FIR channels data if it was allocated */
-		tdfb_free_delaylines(cd);
+		tdfb_free_delaylines(mod);
 
 		/* Allocate all FIR channels data in a big chunk and clear it */
-		cd->fir_delay = rballoc(SOF_MEM_FLAG_USER, delay_size);
+		cd->fir_delay = mod_balloc(mod, delay_size);
```

Comment:
This line will crash without this fix: https://github.com/thesofproject/sof/pull/103